api/ingest-tokens.go

```diff
@@ -79,29 +79,46 @@ func toIngestToken(data ingestTokenData) *IngestToken {
 }
 
 func (i *IngestTokens) Add(repositoryName string, tokenName string, parserName string) (*IngestToken, error) {
-	var mutation struct {
-		Result struct {
-			IngestToken ingestTokenData
-		} `graphql:"addIngestToken(repositoryName: $repositoryName, name: $tokenName, parser: $parserName)"`
+	var parserIdArg *graphql.String
+	if parserName != "" {
+		parser, err := i.client.Parsers().Get(repositoryName, parserName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to look up parser id for parser name %q: %w", parserName, err)
+		}
+		parserIdArg = graphql.NewString(graphql.String(parser.Id))
```

Comment:
if you move `variables` up in the beginning, you can replace this with 
```suggestion
		variables["parserId"] = graphql.String(parser.Id)
```

and avoid the potential nil pointer, might be a little nicer.